node/src/components/linear_chain_sync.rs

```diff
@@ -778,18 +799,43 @@ where
             Event::NewPeerConnected(peer_id) => {
                 trace!(%peer_id, "new peer connected");
                 // Add to the set of peers we can request things from.
-                let mut effects = Effects::new();
-                if self.peers.is_empty() {
-                    // First peer connected, start downloading.
-                    let cloned_peer_id = peer_id.clone();
-                    effects.extend(
-                        effect_builder
-                            .immediately()
-                            .event(move |_| Event::Start(cloned_peer_id)),
-                    );
+                self.peers.push(peer_id.clone());
+                match self.init_hash {
+                    Some(init_hash) if !self.started => {
+                        self.started = true;
+                        let chainspec = self.chainspec.clone();
+                        (async move {
+                            match operations::run_fast_sync_task(
```

Comment:
Here is the entry point for the fast sync task.

I intend to remove the other events for this component.